dwst/scripts/test/particles.js

```diff
@@ -13,10 +13,23 @@
 
 */
 
-import {parseParticles, escapeForParticles, InvalidParticles} from '../particles.js';
+import {parseParticles, escapeForParticles, InvalidParticles, invalidToMlog, finishInvalidParticle} from '../particles.js';
 import {expect} from 'chai';
 
 describe('particles module', () => {
+  describe('invalidToMlog function', () => {
+    it('should return an mlog formatted error message', () => {
+      expect(invalidToMlog(finishInvalidParticle(...[
```

Comment:
don't use `finishInvalidParticle` here just give all the values directly to the constructor since we already know them

Reply:
I am having trouble designing the constructor. I am trying to mix two use cases in the same constructor.

Reply:
Start with the most stupid possible constructor that just takes all the values. You can use null values if you don't have them available and later replace the null values with real values.